ludwig/schema/hyperopt/utils.py

```diff
@@ -0,0 +1,167 @@
+
+
+@DeveloperAPI
+def get_scheduler_dependencies(name: str) -> List[str]:
+    """Get the list of dependencies for a registered hyperopt scheduler.
+
+    Args:
+        name: the name of a scheduler config class registered in `ludwig.schema.hyperopt.scheduler`
+
+    Returns:
+        The list of imports needed to use the scheduler
+    """
+    return scheduler_dependencies_registry[name]
+
+
+@DeveloperAPI
+def get_search_algorithm_cls(name: str) -> Type["BaseSearchAlgorithmConfig"]:  # noqa: F821
+    """Get a registered hyperopt search algorithm config class by name.
+
+    Args:
+        name: the name of a search algorithm config class registered in `ludwig.schema.hyperopt.search_algorithm`
+
+    Returns:
+        A scheduler config class from `ludwig.schema.hyperopt.search_algorithm`
+    """
+    return search_algorithm_registry[name]
+
+
+@DeveloperAPI
+def get_search_algorithm_dependencies(name: str) -> List[str]:
+    """Get the list of dependencies for a registered hyperopt search algorithm.
+
+    Args:
+        name: the name of a search algorithm config class registered in `ludwig.schema.hyperopt.search_algorithm`
+
+    Returns:
+        The list of imports needed to use the search algorithm
+    """
+    return sa_dependencies_registry[name]
+
+
+@DeveloperAPI
+def get_search_algorithm_random_state_field(name: str):
+    """Get the field name of the random state for a registered hyperopt search algorithm.
+
+    Args:
+        name: the name of a search algorithm config class registered in `ludwig.schema.hyperopt.search_algorithm`
+
+    Returns:
+        The name of the random state field in the config
+    """
+    return sa_random_state_field_registry[name]
+
+
+@DeveloperAPI
+def register_parameter_config(name: str) -> Callable:
+    """Register a parameter config class by name.
+
+    Args:
+        name: the name to register the parameter class under, does not need to correspond to the value of `space`
+
+    Returns:
+        Wrapper function to decorate a `BaseParameterConfig` subclass
+    """
+
+    def wrap(cls: Type["BaseParameterConfig"]) -> Type["BaseParameterConfig"]:  # noqa: F821
+        """Add a parameter config class to the registry.
+
+        Args:
+            cls: a subclass of `BaseParameterConfig`
+
+        Returns:
+            `cls` unaltered
+        """
+        parameter_type_registry[name] = cls
+        return cls
+
+    return wrap
+
+
+@DeveloperAPI
+def register_scheduler_config(name: str, dependencies: Optional[List[str]] = None):
+    """Register a scheduler config class by name.
+
+    Args:
+        name: the name to scheduler the parameter class under, does not need to correspond to the value of `type`
+        dependencies: the list of module names that the scheduler requires
+
+    Returns:
+        Wrapper function to decorate a `BaseSchedulerConfig` subclass
+    """
+
+    def wrap(scheduler_config: Type["BaseSchedulerConfig"]) -> Type["BaseSchedulerConfig"]:  # noqa: F821
```

Comment:
Does it just use FIFO in this case as the default?

Reply:
Yes, FIFO is the default.